### PR TITLE
Added a ton of CLI options, plus fixed a CI bug.

### PR DESCRIPTION
**Pull request by TimSimpsonR**
- Renamed the auth_type "basic" to the more apt "auth1.1".
- Made it possible to pass an "token" and "service_url" argument alone to
  the client. It wouldn't work with just this before.
- The client now saves all arguments you give it to the pickled file,
  including the auth strategy, and preserves the token and service_url
  (which it didn't before) which makes exotic auth types such as "fake"
  easier to work with.
- Not raising an error for a lack of an auth_url until auth occurs
  (which is usually right after creation of the client anyway for most
  auth types).
- Moved oparser code into CliOption class. This is where the options live
  plus is the name of that pickled file that gets stored on login.
- Added a "debug" option which avoids swallowing stack traces if
  something goes wrong with the CLI. Should make client work much easier.
- Added a "verbose" option which changes the output to instead show the
  simulated CURL statement plus the request and response headers and
  bodies, which is useful because I...
- Added an "xml" option which does all the communication in XML.
- Fixed a bug which was affecting the CI tests where the client would fail
  if the response body could not be parsed.

### CODE REVIEW

File: reddwarfclient/common.py [hunk 1]
```diff
@@ -123,15 +175,51 @@ def __str__(self):
 class CommandsBase(object):
     params = []
 
+    def __init__(self, parser):
+        self._parse_options(parser)
+
+    #def _get_client(self):
```
Comment (ed-) [inline reddwarfclient/common.py]: bzzt

**Comment by ed-** [comment]
# +1

**Comment by TimSimpsonR** [comment]
Whoops, I messed up the mgmt cli module. I'll need to fix it... one moment please.

**Comment by TimSimpsonR** [comment]
Funny story, the mgmt cli module was completely screwed up before this PR. It's fixed now and working.
Please +1.

**Comment by rnirmal** [comment]
+1

**Comment by cp16net** [comment]
+1

**Pull request merged**